xonsh/built_ins.py

```diff
@@ -569,25 +597,44 @@ def run_subproc(cmds, captured=True):
             else:
                 e = 'Expected callable with 2 or 4 arguments, not {}'
                 raise XonshError(e.format(numargs))
+            if usetee:
+                _tee_file = tempfile.NamedTemporaryFile()
+                tproc = Popen(['tee', _tee_file.name],
+                              stdin=subprocess.PIPE,
+                              stdout=stdout)
+                so = tproc.stdin
+            else:
+                so = stdout
             proc = cls(aliased_cmd, cmd[1:],
-                       stdin, stdout, stderr,
+                       stdin, so, stderr,
                        universal_newlines=uninew)
         else:
             prev_is_proxy = False
-            usetee = (stdout is None) and (not background) and \
-                     ENV.get('XONSH_STORE_STDOUT', False)
-            cls = TeePTYProc if usetee else Popen
             subproc_kwargs = {}
-            if ON_POSIX and cls is Popen:
+            if ON_POSIX:
                 subproc_kwargs['preexec_fn'] = _subproc_pre
             try:
-                proc = cls(aliased_cmd,
-                           universal_newlines=uninew,
-                           env=ENV.detype(),
-                           stdin=stdin,
-                           stdout=stdout,
-                           stderr=stderr,
-                           **subproc_kwargs)
+                if usetee:
+                    _tee_file = tempfile.NamedTemporaryFile()
+                    m, s = pty.openpty()
+                    _set_pty_size(m)
+                    try:
+                        old_sigwinch_handler = signal.signal(signal.SIGWINCH, _sigwinch_maker(m))
+                    except:
+                        pass
+                    tproc = Popen(['tee', _tee_file.name],
```

Comment:
Or is this a tee-only process that then you PIPE the `aliased_cmd` process through?